server/routes/personalRouter.ts

```diff
@@ -72,7 +81,9 @@ export default function personalRouter(services: Services): Router {
     path,
     edit,
     submit,
+    permissionsCheck = editProfileChecks,
```

Comment:
Default to the general edit profile checks, but allow this to be swapped for the diet/allergy version